### PR TITLE
do not throw errors, as mongodb client would auto-reconnect by default

### PR DESCRIPTION
**Pull request by lk9100**

### CODE REVIEW

**Comment by halfninja** [comment]
Need to recreate this PR as a merge into the upstream project

File: lib/agenda.js [hunk 1]
```diff
@@ -607,7 +600,7 @@ function processJobs(extraJob) {
     self._nextScanAt = new Date(now.valueOf() + self._processEvery);
     self._findAndLockNextJob(name, definitions[name], function(err, job) {
       if (err) {
-        throw err;
+        console.log('error with mongodb connection', err);
```
Comment (halfninja) [inline lib/agenda.js:603]: Console.log probably isn't appropriate, and we don't know at this point that it's definitely a mongo error.. could have been passed any error

File: lib/agenda.js [hunk 2]
```diff
@@ -420,33 +420,26 @@ Agenda.prototype._findAndLockNextJob = function(jobName, definition, cb) {
   var self = this,
       now = new Date(),
       lockDeadline = new Date(Date.now().valueOf() - definition.lockLifetime);
-
-  // Don't try and access Mongo Db if we've lost connection to it. Also see clibu_automation.js db.on.close code. NF 29/04/2015
-  // Trying to resolve crash on Dev PC when it resumes from sleep.
   var s = this._mdb.s || this._mdb.db.s;
-  if (s.topology.connections().length === 0) {
```
Comment (halfninja) [inline lib/agenda.js:427]: I think we had trouble with the scheduler restarting when we took this away. It seemed to work best when we just stopped throwing the error, though I don't know why

File: lib/agenda.js [hunk 3]
```diff
@@ -420,33 +420,26 @@ Agenda.prototype._findAndLockNextJob = function(jobName, definition, cb) {
   var self = this,
       now = new Date(),
       lockDeadline = new Date(Date.now().valueOf() - definition.lockLifetime);
-
-  // Don't try and access Mongo Db if we've lost connection to it. Also see clibu_automation.js db.on.close code. NF 29/04/2015
-  // Trying to resolve crash on Dev PC when it resumes from sleep.
   var s = this._mdb.s || this._mdb.db.s;
```
Comment (halfninja) [inline lib/agenda.js:423]: Wouldn't need this line

File: lib/agenda.js (hunk 2 above)
Comment (lk9100) [reply lib/agenda.js:427]: ok will keep this bit..

**Pull request closed without merging**